internal/topology/upgrade/clusterctl_upgrade_test.go

```diff
@@ -121,7 +125,12 @@ func TestDropDefaulterRemoveUnknownOrOmittableFields(t *testing.T) {
 	t.Log("check cluster1 is stable")
 	var cluster1Refs map[clusterv1.ContractVersionedObjectReference]int64
 	g.Eventually(func() error {
-		cluster1Refs, err = getClusterTopologyReferences(cluster1, "v1beta1", checkOmittableField(omittableFieldsMustNotBeSet)) // The defaulting webhook drops the omittable field when using v1beta1.
+		cluster1Refs, err = getClusterTopologyReferences(cluster1, "v1beta1",
+			checkOmittableFromPatchesField(omittableFieldsMustNotBeSet), // The defaulting webhook drops the omittable field when using v1beta1.
+			checkPtrTToT(),                   // "" should never show up in the yaml ("" is not a valid value).
+			checkTToPtrT(),                   // zero or false should never show up due to omitempty.
+			checkTToOmitZeroT(zeroMustBeSet), // zero value should show up (no omitzero).
```

Comment:
Maybe just to make this easier to ready replace T with Type, e.g. checkPtrTypeToType